server/connector/data_sink.cpp

```diff
@@ -92,25 +96,43 @@ void RocksDBDataSink::appendData(velox::RowVectorPtr input) {
   SDB_ASSERT(input->type()->size() == _column_ids.size(),
              "RocksDBDataSink: column oids size ", _column_ids.size(),
              " doesn't match input type size ", input->type()->size());
-  // TODO(Dronplane) implement updating PK fields
-  _row_keys.clear();
-  primary_key::Create(*input, _key_childs, _row_keys);
   SDB_ASSERT(input->type()->kind() == velox::TypeKind::ROW);
-  const auto num_columns = input->childrenSize();
-  std::string table_key = key_utils::PrepareTableKey(_object_key);
-  const auto parent_size = table_key.size();
-  velox::IndexRange all_rows(0, input->size());
-  [[maybe_unused]] const auto& input_type = input->type()->asRow();
-  for (const auto& key : _row_keys) {
-    basics::StrResize(table_key, parent_size);
-    key_utils::AppendPrimaryKey(table_key, key);
+
+  // TODO(Dronplane) implement updating PK fields
+  std::string combined_key = key_utils::PrepareTableKey(_object_key);
+  SDB_ASSERT(sizeof(ObjectId) == combined_key.size());
+  static constexpr auto kObjectKeySize = sizeof(ObjectId);
```

Comment:
Could it be just
```suggestion
  const auto kObjectKeySize = combined_key.size();
```
So we would remove here assumption about building Object Key.
constexpr would possibly have some benefit for memcpy below, but anyway I suggest to move that logic into key_utils where it could be constexpr.